_posts/2021-09-06-connector-table-sql-api-part2.md

```diff
@@ -0,0 +1,519 @@
+---
+layout: post
+title:  "Implementing a custom source connector for Table API and SQL - Part Two "
+date: 2021-09-06T00:00:00.000Z
+authors:
+- Ingo Buerk:
+  name: "Ingo Buerk"
+- Daisy Tsang:
+  name: "Daisy Tsang"
+excerpt: 
+---
+
+{% toc %}
+
+# Introduction
+
+In [part one](/2021/09/06/connector-table-sql-api-part2) of this tutorial, you learned how to build a custom source connector for Flink. In part two, you will learn how to integrate the connector with a test email inbox through the IMAP protocol, filter out emails, and execute [Flink SQL on the Ververica Platform](https://www.ververica.com/apache-flink-sql-on-ververica-platform). 
+
+# Goals
+
+Part two of the tutorial will teach you how to: 
+
+- integrate a source connector which connects to a mailbox using the IMAP protocol
+- use [Jakarta Mail](https://eclipse-ee4j.github.io/mail/), a Java library that can send and receive email via the IMAP protocol  
+- write [Flink SQL](https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/dev/table/sql/overview/) and execute the queries in the Ververica Platform
+
+You are encouraged to follow along with the code in this [repository](https://github.com/Airblader/blog-imap). It provides a boilerplate project that also comes with a bundled [docker-compose](https://docs.docker.com/compose/) setup that lets you easily run the connector. You can then try it out with Flink’s SQL client.
+
+
+# Prerequisites
+
+This tutorial assumes that you have:
+
+- followed the steps outlined in [part one](/2021/09/06/connector-table-sql-api-part1) of this tutorial
+- some familiarity with Java and objected-oriented programming
+
+
+# Understand how to fetch emails via the IMAP protocol
+
+Now that you have a working source connector that can run on Flink, it is time to connect to an email server via [IMAP](https://en.wikipedia.org/wiki/Internet_Message_Access_Protocol) (an Internet protocol that allows email clients to retrieve messages from a mail server) so that Flink can process emails instead of test static data.  
+
+You will use [Jakarta Mail](https://eclipse-ee4j.github.io/mail/), a Java library that can be used to send and receive email via IMAP. For simplicity, authentication will use a plain username and password.
+
+This tutorial will focus more on how to implement a connector for Flink. If you want to learn more about the details of how IMAP or Jakarta Mail work, you are encouraged to explore a more extensive implementation at this [repository](https://github.com/TNG/flink-connector-email). 
```

Comment:
mention what is so extensive about it by some examples. this repo is very useful and deserves more attention than one sentence. in the second setence we can list some abilities that it implements.